test/helpers/integration.rb

```diff
@@ -36,26 +37,31 @@ def teardown
       io.close if io.is_a?(IO) && !io.closed?
       io = nil
     end
+    File.unlink(@bind_path) rescue nil
   end
 
   private
 
-  def cli_server(argv, bind = nil)
-    if bind
-      cmd = "#{BASE} bin/puma -b #{bind} #{argv}"
+  def cli_server(argv, unix: false)
+    if unix
+      cmd = "#{BASE} bin/puma -b unix://#{@bind_path} #{argv}"
     else
       @tcp_port = UniquePort.call
       cmd = "#{BASE} bin/puma -b tcp://#{HOST}:#{@tcp_port} #{argv}"
     end
     @server = IO.popen(cmd, "r")
     wait_for_server_to_boot
+    @pid = @server.pid
     @server
   end
 
-  def send_term_to_server(pid)
-    Process.kill(:TERM, pid)
+  def stop_server(pid = @pid, signal: :TERM)
+    Process.kill signal, pid
     sleep 1
-    Process.wait2(pid)
+    begin
+      Process.wait2 pid
+    rescue Errno::ECHILD
```

Comment:
Shouldn't we just wait2 without sleeping, then we don't need to rescue ECHILD?

Reply:
At least one test is checking an exit code:

https://github.com/puma/puma/blob/17c1d2997e7d78dcda3f25e72baa2bc29fb6d036/test/test_integration_cluster.rb#L60-L65.  Shouldn't the wait be in place for that?

Reply:
wait2 waits for the pid to exit, then returns the exit code. I don't see why a sleep or rescue is necessary.

Reply:
Ok.  The sleep should be able to be removed, not sure if the rescue can be.  I'll remove both.

Reply:
@MSP-Greg ^^^^

Reply:
Just pushed an update.